### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by b3an5**
@thatPamIAm
@brittanystoroz
@robbiejaeger

this pull request parsed through the data to get 4 random categories and displaying them in the top 4 boxes, we are splicing them out of the 'full' data set and put them into a 'in use data set' to use during the current round. by splicing them out we assure that we wont have to worry about re-using a category during the game in other rounds. we need to do some refactoring because we probably should make a new method who's sole job is to parse through the dataset. I would like some feedback from Gabe as to where we should place this method and or any critiques that you have.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,3 +59,5 @@ typings/
 
 # next.js build output
 .next
+
+.Ds_Store
```
Comment (brittanystoroz) [inline .gitignore]: it's `.DS_Store` with a capital 'S' ;)

File: JS/Board.js [hunk 2]
```diff
@@ -6,9 +6,36 @@ class Board {
     this.completed = false
   }
 
+
+  /////this needs refactoring
```
Comment (brittanystoroz) [inline JS/Board.js]: File an issue for something that needs refactoring rather than leaving it as a comment in the codebase -- if you want your partner to be aware of the fact that "maybe this is still a little sloppy", just put that  in the description of your PR or leave a line comment here like I'm doing with this review.

**Comment by brittanystoroz** [comment]
Really nice-sized pull request, readable with small chunks of code though the PR title is a bit vague and you should keep irrelevant changesets out of here (like adding .DS_Store to the .gitignore file)

File: JS/Board.js [hunk 3]
```diff
@@ -6,9 +6,36 @@ class Board {
...
   populateBoard() {
     // runs dom fn
-    //gets values from dataset
+    let clues = (Object.values(data)[2])
+    console.log('clues', clues)
```
Comment (brittanystoroz) [inline JS/Board.js]: Meep meep, also leave out the `console.logs` -- those are useful for you, while you're developing, but just add clutter to the diff for your partner to read through.

**Comment by brittanystoroz** [comment]
I will let @Inzurriaga chime in on the code before I add any more line comments, but plan on some extra notes from me before merging this :)

**Comment by b3an5** [comment]
@brittanystoroz @Inzurriaga  and I just worked on this new push let me know what you think

File: JS/index.js [hunk 4]
```diff
@@ -1 +1,19 @@
...
+
+$('.start-button').on('click', (e) => {
+  e.preventDefault()
+  $('.form').css('display', 'none');
+  $('#tv-frame').css('animation', 'tvframe 5s')
+  game.init();
+  setTimeout(() => {
+    $('#splash-page').css('display', 'none')
+  }, 2000)
+  $('.hidden').css('display', 'block')
```
Comment (brittanystoroz) [inline JS/index.js]: There should be a jQuery short-hand for the display none/block --- I think you can do something like `$(el).show()` and $(el).hide()` instead.

File: JS/Game.js [hunk 5]
```diff
@@ -1,14 +1,39 @@
...
+    domUpdates.displayNames()
+    let playerOneName = $('.js-player1-name-input').val()
+    let playerTwoName = $('.js-player2-name-input').val()
+    let playerThreeName = $('.js-player3-name-input').val()
+    let playerOne = new Player(playerOneName);
+    this.players.push(playerOne)
+    let playerTwo = new Player(playerTwoName);
+    this.players.push(playerTwo)
+    let playerThree = new Player(playerThreeName);
+    this.players.push(playerThree)
```
Comment (brittanystoroz) [inline JS/Game.js]: This looks like a good use-case for a `for` loop. Run the for loop three times and you can cut this down to two lines of code. Something like:

```
let player = $(`.js-player${i}-name-input`).val();
this.players.push(player);
```
```

File: JS/Game.js [hunk 6]
```diff
@@ -1,14 +1,39 @@
 class Game {
   constructor() {
     this.round = 0,
-    this.players = {}, // {player1: 'whatever is put in imput', ...}
+    this.players = [], // {player1: 'whatever is put in imput', ...}
     this.round3 = false,
     this.winner = null
+    this.data;
```
Comment (brittanystoroz) [inline JS/Game.js]: Is there no default value for the data on your Game class? What is this ultimately going to represent?

File: JS/Dom.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  //   $('#js-player1-score')
+  //   $('#js-player2-score')
+  //   $('#js-player3-score')
+  // }
+
+  displayCat(cats) {
+    $('#one').text(cats[0]);
+    $('#two').text(cats[1]);
+    $('#three').text(cats[2]);
+    $('#four').text(cats[3]);
```
Comment (brittanystoroz) [inline JS/Dom.js]: Numbers for ids are usually not informative enough. What are you counting? Additionally, if you need to number your ids, it's probably a sign you should be using a class instead -- it seems like you have 4 elements that represent a category.  We could select the category elements by a class instead, gain access to all 4 of them, loop over them and assign their text values with their corresponding category name.

File: JS/Game.js [hunk 8]
```diff
@@ -1,14 +1,39 @@
...
+    this.players.push(playerTwo)
+    let playerThree = new Player(playerThreeName);
+    this.players.push(playerThree)
+    this.round++
+  }
+
+  parseData() {
+    let clues = (Object.values(data)[2])
+
+    this.data = (Object.keys(Object.values(data)[1])).map((category, index) => {
```
Comment (brittanystoroz) [inline JS/Game.js]: This is a little hard to follow...there's a bit too much going on with `Object.keys(Object.values(data)[1]))`

Can we break these out into variables that have semantic names to help me follow it a bit better? What does `data[1]` represent? What does `Object.values()` for that piece of information represent? What does `Object.keys()` for that information give us? Category Names?

File: JS/Game.js [hunk 9]
```diff
@@ -1,14 +1,39 @@
...
+      let clueVal = []
+      clues.forEach((clue) => {
+        if (clue.categoryId === index + 1) {
+          clueVal.push(clue)
+        }
+      })
+      return { category, clues: clueVal }
```
Comment (brittanystoroz) [inline JS/Game.js]: Nice shorthand with the `category` property

File: JS/Game.js [hunk 10]
```diff
@@ -1,14 +1,39 @@
...
+  parseData() {
+    let clues = (Object.values(data)[2])
+
+    this.data = (Object.keys(Object.values(data)[1])).map((category, index) => {
+      let clueVal = []
+      clues.forEach((clue) => {
+        if (clue.categoryId === index + 1) {
+          clueVal.push(clue)
+        }
+      })
```
Comment (brittanystoroz) [inline JS/Game.js]: Not sure here, but it almost looks like you're re-implementing the functionality of `filter` here? You're building up an array that's going to be a subset of your original `clues` array based on a condition in the callback of `forEach`...

File: JS/Game.js (hunk 6 above)
Comment (b3an5) [reply JS/Game.js]: Game.data becomes equal to our newly structured dataset
